### PR TITLE
iTwin.js Docs Release Pipeline

### PR DESCRIPTION
**Pull request by skirby1996**
Adds a YAML based to pipeline to release the iTwin.js Docs website.

Currently waiting on:
- [x] [Update scorecard validators to support release of static sites](https://dev.azure.com/bentleycs/iModelTechnologies/_git/imodeljs-build-pipeline-scripts/pullrequest/293613)
- [ ] [Update package versions for docs selenium tests](https://dev.azure.com/bentleycs/iModelTechnologies/_git/BeMetalsmith/pullrequest/294787)

### CODE REVIEW

**Review by pmconne — APPROVED** [review]
Approving typo fix.

File: common/config/azure-pipelines/templates/gather-docs.yaml [hunk 1]
```diff
@@ -69,8 +71,8 @@ steps:
     inputs:
       buildType: specific
       project: '2c48216e-e72f-48b4-a4eb-40ff1c04e8e4'
-      definition: 1734
+      definition: 7436 # iTwin.js/Docs/iTwin.js Docs - YAML
       allowPartiallySucceededBuilds: true
-      buildVersionToDownload: latestFromBranch
+      buildVersionToDownload: latest
```
Comment (calebmshafer) [inline common/config/azure-pipelines/templates/gather-docs.yaml:76]: I think we'll need latestFrom branch to make it work correctly on release branches right?

File: common/config/azure-pipelines/jobs/docs/docs-build.yaml [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+    steps:
+      - checkout: self
+        path: itwinjs-core
+        clean: true
+
+      - task: NodeTool@0
+        displayName: Use Node 14
+        inputs:
+          versionSpec: 14.17.4
```
Comment (calebmshafer) [inline common/config/azure-pipelines/jobs/docs/docs-build.yaml:37]: Can we update this to Node 18 or is there some restriction?

Comment (skirby1996) [reply common/config/azure-pipelines/jobs/docs/docs-build.yaml:37]: I tried bumping up to node 16 everywhere in the docs pipeline, but that started causing test failures so I just reverted back to node 14 everywhere. I think I could probably update the node version everywhere except the docs product tests without issue, but not sure if we want to keep our node version consistent throughout or not.

File: common/config/azure-pipelines/templates/gather-docs.yaml (hunk 1 above)
Comment (skirby1996) [reply common/config/azure-pipelines/templates/gather-docs.yaml:76]: I'll fix this, we'll also need to make sure we set the `branchName` value accordingly for release branches (the default is 'refs/heads/master' so we're OK here.) I'll make sure to add a note to the docs and make add a backlog item to automate this when we create the release branch moving forward.

I did want to come back to this, I think it makes more sense to pull the .updated.json from the release pipeline and not the PR validation pipeline, but we need populate the new release pipeline with a few successful builds from the actual master and release branches first. I'd also planned on turning off triggers for pushes to a branch in the PR validation pipeline when I did this too.

File: common/config/azure-pipelines/jobs/docs/docs-build.yaml (hunk 2 above)
Comment (calebmshafer) [reply common/config/azure-pipelines/jobs/docs/docs-build.yaml:37]: Okay, let's leave it as is for now and then follow up in a separate PR.

**Review by calebmshafer — APPROVED** [review]

**Pull request merged**